lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc

```diff
@@ -382,87 +382,132 @@ function paraneue_dosomething_preprocess_node_campaign(&$vars) {
     drupal_add_js($js, 'inline');
   }
 
+  // Setup the reportback showcase.
+  paraneue_dosomething_preprocess_reportback_showcase($vars);
+
   // If this is not the pitch page:
   if (!isset($vars['is_pitch_page'])) {
     // Set the number of files to display in initial gallery.
-    $vars['reportbacks']['initial_count'] = 6;
+    $vars['reportbacks_gallery']['desired_count'] = 6;
     // Adds themed Reportback Files.
-    paraneue_dosomething_preprocess_reportback_files($vars);
+    paraneue_dosomething_preprocess_reportback_gallery($vars);
   }
 
 }
 
+
 /**
- * Adds $gallery_items array of themed Reportback Files for given node preprocess $vars.
+ * Get total number of Reportbacks by status (promoted, approved, etc).
+ *
+ * @param $nid  Campaign node id.
+ * @param $status Reportback status (promoted, approved, etc).
+ * @return int
  */
-function paraneue_dosomething_preprocess_reportback_files(&$vars) {
-  $reportbacks_gallery = &$vars['reportbacks_gallery'];
+function paraneue_dosomething_preprocess_reportbacks_total_by_status($nid, $status) {
+  $params = array();
+  $params['nid'] = $nid;
+  $params['status'] = $status;
 
-  if (!isset($reportbacks_gallery['initial_count'])) {
-    $reportbacks_gallery['initial_count'] = 6;
-  }
-  $initial_count = $reportbacks_gallery['initial_count'];
+  return dosomething_reportback_get_reportback_files_query_count($params);
+}
 
-  $reportbacks_gallery['items'] = array();
 
+/**
+ * Get a specified number of themed Promoted Reportbacks.
+ *
+ * @param $nid  Campaign node id.
+ * @param $count Number of Promoted Reportbacks to retrieve.
+ * @return array
+ */
+function paraneue_dosomething_preprocess_themed_promoted_reportbacks($nid, $count, $random = FALSE) {
```

Comment:
Same here, usually a preprocess function takes variables by reference and messes with them.  I'd recommend just renaming this as `paraneue_dosomething_get_themed_promoted_reportbacks` ?